### PR TITLE
[Zero-Dim] support input 0D Tensor for sundary api

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

为无需测反向的这类API支持输入0D Tensor，例如**判断类、属性**等API，如下：
---

```
paddle.nn.functional.linear
paddle.is_complex
paddle.is_floating_point
paddle.is_integer
paddle.is_tensor
paddle.isfinite
paddle.isinf
paddle.isnan
paddle.isclose
paddle.clone
paddle.assign
paddle.Tensor.item
paddle.Tensor.tolist
paddle.Tensor.numpy
paddle.numel
paddle.rank
paddle.shape
```

### CODE REVIEW

**Review by XieYunshen — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 1]
```diff
@@ -232,31 +234,32 @@ def test_static(self):
...
-    paddle.pow,
-]
-
-binary_api_list_without_grad = [
+    {'func': paddle.pow, 'cls_method': '__pow__'},
     {'func': paddle.add, 'cls_method': '__add__'},
     {'func': paddle.subtract, 'cls_method': '__sub__'},
     {'func': paddle.multiply, 'cls_method': '__mul__'},
     {'func': paddle.divide, 'cls_method': '__div__'},
     {'func': paddle.subtract, 'cls_method': '__sub__'},
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: why did `paddle.subtract` repeat 4 times, while `paddle.add` 2 times, what is the meaning of repeat in the list?

Comment (jeff41404) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: shall we add `paddle.maximum` and `paddle.minimum` ?

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -232,31 +234,32 @@ def test_static(self):
...
     {'func': paddle.greater_equal, 'cls_method': '__ge__'},
     {'func': paddle.greater_than, 'cls_method': '__gt__'},
     {'func': paddle.less_equal, 'cls_method': '__le__'},
     {'func': paddle.less_than, 'cls_method': '__lt__'},
+    {'func': paddle.remainder, 'cls_method': '__mod__'},
+    paddle.mod,
+    paddle.floor_mod,
     paddle.logical_and,
     paddle.logical_or,
     paddle.logical_xor,
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:251]: shall we add `paddle.bitwise_and/or/xor` ?

Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:251]: 已添加

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (zhouwei25) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: Some mistake, have fixed it. thx~

Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: > 

目前只支持了部分二元，就是这个list中包含的。这两个还未支持0D，后面会加。

Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: > why did `paddle.subtract` repeat 4 times, while `paddle.add` 2 times, what is the meaning of repeat in the list?

Some mistake, have fixed it.

**Review by XieYunshen — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (jeff41404) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: ok, thanks
